### PR TITLE
Initial implementation of `VmCodeDisplay` in program explorer

### PR DESCRIPTION
**Pull request by bkonyi**
- Adds support for displaying code objects in the `ProgramExplorer` widget
- Adds initial implementation for displaying code objects in the `VM Tools` `Objects` view.

**Screenshot:**
Screen Shot 2022-07-22 at 12 25 26 PM

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/vm_developer/vm_service_private_extensions.dart [hunk 1]
```diff
@@ -70,3 +75,109 @@ extension ClassHeapStatsPrivateViewExtension on ClassHeapStats {
...
+  /// The list of [Instructions] that make up the generated code.
+  final instructions = <Instruction>[];
+
+  List toJson() => [
+        for (final i in instructions) ...i.toJson(),
+      ];
+}
+
+/// An extension on [Func] which allows for access to VM internal fields.
+
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/vm_developer/vm_service_private_extensions.dart:164]: nit remove blank line

File: packages/devtools_app/lib/src/screens/vm_developer/vm_service_private_extensions.dart [hunk 2]
```diff
@@ -70,3 +75,109 @@ extension ClassHeapStatsPrivateViewExtension on ClassHeapStats {
...
+extension FunctionPrivateViewExtension on Func {
+  static const _unoptimizedCodeKey = '_unoptimizedCode';
+
+  /// The unoptimized [CodeRef] associated with the [Func].
+  CodeRef? get unoptimizedCode => CodeRef.parse(json![_unoptimizedCodeKey]);
+  set unoptimizedCode(CodeRef? code) => json![_unoptimizedCodeKey] = code?.json;
+}
+
+/// An extension on [Code] which allows for access to VM internal fields.
+
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/vm_developer/vm_service_private_extensions.dart:174]: same

File: packages/devtools_app/macos/Runner.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -395,7 +395,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				MACOSX_DEPLOYMENT_TARGET = 10.11;
```
Comment (kenzieschmoll) [inline packages/devtools_app/macos/Runner.xcodeproj/project.pbxproj:398]: I think these changes should go away once you merge with master

**Review by kenzieschmoll — APPROVED** [review]
some nits and questions but overall lgtm

File: packages/devtools_app/test/vm_developer/object_inspector/vm_code_display_test.dart [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  late MockCodeObject mockCodeObject;
+
+  const windowSize = Size(4000.0, 4000.0);
+
+  group('VmCodeDisplay', () {
+    setUpAll(() {
+      setGlobal(IdeTheme, IdeTheme());
+
+      mockCodeObject = MockCodeObject();
+      final testCode = Code(
```
Comment (migueltylee) [inline packages/devtools_app/test/vm_developer/object_inspector/vm_code_display_test.dart:27]: Should testCode be in vm_developer_test_utils.dart?

**Review by migueltylee — COMMENTED** [review]
LGTM!

File: packages/devtools_app/lib/src/screens/vm_developer/vm_code_display.dart [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+
+  @override
+  int getValue(Instruction dataObject) {
+    return int.parse(dataObject.address, radix: 16);
+  }
+
+  @override
+  String getDisplayValue(Instruction dataObject) {
+    final value = getValue(dataObject);
+    return '0x${value.toRadixString(16).toUpperCase().padLeft(8)}';
```
[the remark this answers is not in the document]
Comment (bkonyi) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_code_display.dart:42]: To make sure smaller addresses don't get truncated (e.g., `0x8` should be `0x00000008`)

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_tools_screen.dart [hunk 6]
```diff
@@ -14,7 +14,7 @@ import 'object_inspector_view.dart';
 import 'vm_developer_tools_controller.dart';
 import 'vm_statistics_view.dart';
 
-const displayObjectInspector = false;
+const displayObjectInspector = true;
```
[the remark this answers is not in the document]
Comment (bkonyi) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_developer_tools_screen.dart:17]: Nope... :-)

File: packages/devtools_app/lib/src/screens/vm_developer/vm_object_model.dart [hunk 7]
```diff
@@ -144,3 +144,15 @@ class InstanceObject extends VmObject {
   @override
   String? get name => obj.name;
 }
+
+class CodeObject extends VmObject {
+  CodeObject({required super.ref});
```
[the remark this answers is not in the document]
Comment (bkonyi) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_object_model.dart:149]: Yeah, we can do that when we get around to merging your change.

File: packages/devtools_app/test/vm_developer/object_inspector/vm_code_display_test.dart (hunk 4 above)
Comment (bkonyi) [reply packages/devtools_app/test/vm_developer/object_inspector/vm_code_display_test.dart:27]: That's a good question. Maybe once more functionality is written, but I think it's worth leaving here for now.

**Pull request merged**